tests/test_js_prefs.py

```diff
@@ -23,6 +23,17 @@ def test_pref_innocuous_branch():
     pref("extensions.foo-bar.baz", true);
     """, path="defaults/preferences/prefs.js").failed()
 
+def test_unicode_pref():
+    """Tests that non-ASCII preferences do not cause errors."""
+
+    assert not _do_test_raw("""
+    pref("extensions.foo-bar.\u263a", true);
+    """, path="defaults/preferences/prefs.js").failed()
+
+    assert not _do_test_raw("""
+    pref("extensions.foo-bar.\\u263a", true);
+    """, path="defaults/preferences/prefs.js").failed()
+
```

Comment:
Missing the test with `☺` and for good measure try `éç` which is the only one that raised the exception, at least on my computer (even with the patch from #294).